### DISCUSSION TITLE
Check Load balancers for CIDR range

### DISCUSSION THREAD

**Discussion by SteveL1** [cloud-custodian/cloud-custodian#7446] [AWS]
### Ask your question

Has anyone managed to use the security-group type as a filter for another resource (in my case elb and app-elb) to check or CIDRs

The closest I have got is this:
  - name: elb-test-policy
    resource: elb
    filters:
      - type: security-group
        key: "IpPermissions[].IpRanges[].CidrIp"
        value_type: cidr
        op: equal
        value: 
        - 10.0.0.0/8 

But this returns all load balancers regardless of what security groups are attached.

I can't really create a tag on the security group and use this in the filter as we have different policies depending on tags on the load balancer.

Any help would be appreciated, i've book looking at this for hours now!!!

**Comment by ajkerrigan** [comment]
That's possible, if a little awkward at the moment. The sample policy you have looks pretty close. Your `key` there is bringing back a list of CIDR ranges across all rules. If you're looking to see if `10.0.0.0/8` appears in that list, you could use a filter like:

```yaml
- type: security-group
  key: "IpPermissions[].IpRanges[].CidrIp"
  op: intersect
  value:
    - "10.0.0.0/8"
```

When I say "at the moment", it's because this question bumps into some related issues and discussion. There's an open PR (#7129) related to matching against lists of CIDRs - today `value_type: cidr` is focused on resolving a single range at a time. We've also had some discussion about bringing versions of `aws.security-group`'s richer `ingress` and `egress` filters to resources that support a `security-group` filter. You're not alone in looking for this sort of filter!

**Comment by SteveL1** [comment]
Thanks for this, it sounds like it should exactly what I need, but when I try it I add this to my policy I get the error

`
custodian.commands:ERROR ['10.0.0.0/8'] is not of type 'number'
`
